ansible/roles_ocp_workloads/ocp4_workload_cyberark_dap/tasks/workload.yml

```diff
@@ -0,0 +1,134 @@
+---
+- name: Setting up CyberArk DAP workload for user
+  debug:
+    msg: "Setting up CyberArk DAP workload for user ocp_username = {{ ocp_username }}"
+
+# CyberArk DAP workload tasks
+# -------------------------------------------------------------------------
+
+- name: Create the cyberark setup directory
+  file:
+    path: /home/{{ansible_user}}/cyberark_setup
+    state: directory
+
+- name: Download the CyberArk RHPDS setup files
+  become: true
+  unarchive:
+    src: "{{ ocp4_workload_cyberark_setup_files_url }}"
+    dest: /home/{{ansible_user}}/cyberark_setup
+    remote_src: True
+    owner: "{{ ansible_user }}"
+    group: users
+
+- name: Set random password for dap
+  set_fact:
+    dap_pwd: "{{ lookup('password', '/dev/null length=41 chars=ascii_letters,digits,hexdigits,@,#,$,%,^,*,') }}"
+
+- name: create CyberArk resources
+  k8s:
+    state: present
+    definition: "{{ lookup('template', item ) | from_yaml }}"
+  loop:
+    - ./templates/namespace.j2
+    - ./templates/service.j2
+    - ./templates/dap_authn_role.j2
+    - ./templates/role_binding.j2
+    - ./templates/dap_cm_role.j2
+    - ./templates/secrets_cluster_role.j2
+
+- name: Add policy add-scc-to-user dap authn service
+  command: "oc adm policy add-scc-to-user anyuid -z dap-authn-service -n {{ ocp4_workload_cyberark_project }}"
+
+- name: Deploy cyberark conjur master
+  command: "/home/{{ansible_user}}/cyberark_setup/master_deploy.sh {{ ocp4_workload_cyberark_project }} {{ dap_pwd }}"
+
+- name: Deploy mysqldb
+  command: "/home/{{ansible_user}}/cyberark_setup/mysqldb_deploy.sh {{ ocp4_workload_cyberark_project }}"
+
+- name: Create lab user demo projects
+  k8s:
+    state: present
+    definition:
+      apiVersion: project.openshift.io/v1
+      kind: Project
+      metadata:
+        name: 'user{{ item }}'
+  loop: "{{ range(1, num_users | int + 1, 1) | list }}"
+
+- name: Grant the authentication service access to pods in each user project
+  k8s:
+    state: present
+    definition:
+      kind: RoleBinding
+      apiVersion: rbac.authorization.k8s.io/v1
+      metadata:
+        name: dap-authn-service
+        namespace: 'user{{ item }}'
+      subjects:
+        - kind: ServiceAccount
+          name: dap-authn-service
+          namespace: '{{ ocp4_workload_cyberark_project }}'
+      roleRef:
+        kind: ClusterRole
+        name: dap-authn-role
+        apiGroup: rbac.authorization.k8s.io
+  loop: "{{ range(1, num_users | int + 1, 1) | list }}"
+
+- name: Give lab users admin access to their projects
+  command: "oc adm policy add-role-to-user admin user{{ item }} -n user{{ item }}"
```

Comment:
This `oc adm policy` command is not specifying in which namespace to create the RoleBinding. Also, generally better to to just use a `RoleBinding` definition rather than use `oc adm` for RBAC

Reply:
Thanks @jkupferer - the namespace is specified with the -n user{{ item }} param, I believe... but I will look into creating a definition instead.

Reply:
Done, created a definition